src/agents/agents.go

```diff
@@ -0,0 +1,60 @@
+package agents
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cornelk/hashmap"
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift-assisted/assisted-swarm/models"
+)
+
+var agentList = &hashmap.HashMap{}
```

Comment:
Could you add a small comment that explains that we're using cornelk/hashmap instead of regular go maps because they're thread-safe?

Reply:
Would be nicer to use a 1.18 generic lock-free hashmap with actual compile-time type safety but it doesn't seem like those exist yet